pump/pump_progress.go

```diff
@@ -0,0 +1,63 @@
+package pump
+
+import "gopkg.in/cheggaaa/pb.v1"
+
+type ProgressWriter interface {
+	Increment() int
+	SetTotal(total int)
+	Prefix(elem string)
+	Finish()
+}
+
+type ProgressBarWriter struct {
```

Comment:
What about just not have this at all?
`*pb.ProgressBar` implement this interface, which means you can let the final application create the progress bar and configure it however it makes sense there, and pass it to `PumpIt`. Then this library doesn't make UI choices anymore.

Reply:
Funny you say that bc that was my first approach too.

The `pb.ProgressBar` actually doesn't satisfy this interface and I really dislike their way of combining a pointer on all the methods + return the struct itself like:

```
func (pb *ProgressBar) SetTotal(total int) *ProgressBar {
```